src/InputStepper/index.js

```diff
@@ -7,140 +7,143 @@ import InputStepperStateless from "./InputStepperStateless";
-    const prevValue = this.state.value;
+  const handleChange = (ev: SyntheticInputEvent<HTMLInputElement>) => {
+    const { minValue = Number.NEGATIVE_INFINITY, maxValue = Number.POSITIVE_INFINITY } = props;
+    const eventValue = ev && parseInt(ev.target.value, 10);
+    const prevValue = value;
 
-    if (prevValue <= value) {
-      this.setState({ value: validateIncrement({ value, maxValue, step: 0 }) });
+    if (prevValue <= eventValue) {
+      setValue(validateIncrement({ value, maxValue, step: 0 }));
     }
 
-    if (prevValue >= value) {
-      this.setState({ value: validateDecrement({ value, minValue, step: 0 }) });
+    if (prevValue >= eventValue) {
+      setValue(validateDecrement({ value, minValue, step: 0 }));
     }
   };
 
-  render() {
-    const {
-      label,
-      error,
-      onBlur,
-      onFocus,
-      help,
-      disabled,
-      name,
-      dataTest,
-      size = SIZE_OPTIONS.NORMAL,
-      maxValue,
-      minValue,
-      required,
-      tabIndex,
-      forwardedRef,
-      spaceAfter,
-      titleIncrement,
-      titleDecrement,
-    } = this.props;
-    const { value } = this.state;
-    return (
-      <InputStepperStateless
-        dataTest={dataTest}
-        size={size}
-        label={label}
-        disabled={disabled}
-        required={required}
-        name={name}
-        error={error}
-        help={help}
-        onChange={this.handleChange}
-        onBlur={onBlur}
-        onFocus={onFocus}
-        onKeyDown={this.handleKeyDown}
-        value={value || 0}
-        minValue={minValue}
-        maxValue={maxValue}
-        tabIndex={tabIndex}
-        forwardedRef={forwardedRef}
-        spaceAfter={spaceAfter}
-        onDecrement={this.handleDecrementCounter}
-        onIncrement={this.handleIncrementCounter}
-        titleIncrement={titleIncrement}
-        titleDecrement={titleDecrement}
-      />
-    );
-  }
-}
+  const {
+    label,
+    error,
+    onBlur,
+    onFocus,
+    help,
+    disabled,
+    name,
+    dataTest,
+    size = SIZE_OPTIONS.NORMAL,
+    maxValue,
+    minValue,
+    required,
+    tabIndex,
+    forwardedRef,
+    spaceAfter,
+    titleIncrement,
+    titleDecrement,
+  } = props;
+
+  return (
+    <InputStepperStateless
+      dataTest={dataTest}
+      size={size}
+      label={label}
+      disabled={disabled}
+      required={required}
+      name={name}
+      error={error}
+      help={help}
+      onChange={handleChange}
+      onBlur={onBlur}
+      onFocus={onFocus}
+      onKeyDown={handleKeyDown}
+      value={value || 0}
+      minValue={minValue}
+      maxValue={maxValue}
+      tabIndex={tabIndex}
+      forwardedRef={forwardedRef}
+      spaceAfter={spaceAfter}
+      onDecrement={handleDecrementCounter}
+      onIncrement={handleIncrementCounter}
+      titleIncrement={titleIncrement}
+      titleDecrement={titleDecrement}
+    />
+  );
+};
 
 // $FlowExpected
 const ForwardedInputStepper = React.forwardRef((props, ref) => (
```

Comment:
You can forward ref immediately on L17, check `Button` for instance. This legacy code was due to usage of `Class` component.